platform-ajax/src/main/typescript/com/softicar/platform/ajax/engine/request/AjaxRequestMessageEncoder.ts

```diff
@@ -1,29 +1,28 @@
 
 /**
- * This class encodes the {@link AjaxRequest} parameters.
+ * This class encodes the {@link AjaxRequestMessage} data.
  */
-class AjaxMessageEncoder {
+class AjaxRequestMessageEncoder {
 	private static HEX_DIGITS = "0123456789ABCDEF";
-	private parameters: any;
+	private data: Map<string, string>;
 	
-	public constructor(parameters: any) {
-		this.parameters = parameters;
+	public constructor(data: Map<string, string>) {
+		this.data = data;
 	}
 
 	/**
-	 * Encodes the {@link AjaxRequest} parameters into a string.
+	 * Encodes the {@link AjaxRequest} data into a string.
 	 */
 	public encode(): string {
 		let data = [];
 
-		for(let key in this.parameters) {
+		for(let [key, value] of this.data) {
 			// push key
 			data.push(key.length+"\n"+key);
 	
 			// push value
-			let value = this.parameters[key];
+			// FIXME this check should be not needed!!!!!!!!!!!!!!!
```

Comment:
what about this? when/how should we fix it?

Reply:
nice catch, I added this before finishing `AjaxRequestMessage`, now I can say that the check is indeed useless